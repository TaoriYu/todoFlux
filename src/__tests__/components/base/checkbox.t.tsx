import * as React from 'react';
import {shallow} from 'enzyme';
import {Checkbox} from 'semantic-ui-react';
import renderer = require('react-test-renderer');

describe('Checkbox component', () => {
  test('basic snapshot render', () => {
    const checkbox = renderer.create(
      <div>
        <Checkbox>If life is easy then what is this?</Checkbox>
        <Checkbox label="So look into my eyes and see the bull's eye and the crack"/>
      </div>
    );

    let tree = checkbox.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('onclick Checkbox', () => {
    const checkbox = shallow(
      <Checkbox>Afraid to talk but is hard to catch</Checkbox>
    );

    checkbox.simulate('click');
    expect(checkbox.find('.ui.checkbox').props().className).toEqual('ui checked fitted checkbox');
    checkbox.find('.ui.checkbox').simulate('click');
    expect(checkbox.find('.ui.checkbox').props().className).toEqual('ui fitted checkbox');
  });
});
