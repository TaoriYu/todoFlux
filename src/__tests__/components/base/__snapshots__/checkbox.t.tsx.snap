// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component basic snapshot render 1`] = `
<div>
  <div
    className="ui fitted checkbox"
    onChange={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <input
      checked={false}
      className="hidden"
      name={undefined}
      readOnly={true}
      tabIndex={0}
      type="checkbox"
      value={undefined}
    />
    <label />
  </div>
  <div
    className="ui checkbox"
    onChange={[Function]}
    onClick={[Function]}
    onMouseDown={[Function]}
  >
    <input
      checked={false}
      className="hidden"
      name={undefined}
      readOnly={true}
      tabIndex={0}
      type="checkbox"
      value={undefined}
    />
    <label>
      So look into my eyes and see the bull's eye and the crack
    </label>
  </div>
</div>
`;
